### ISSUE TITLE
UnhandledException event fired on promise chain with Catch handler

### ISSUE THREAD

**Issue by theJenix** [Real-Serious-Games/C-Sharp-Promise#54]
I'm hitting a snag that I think might be a bug, but it also may just be a misunderstanding on my part. 

I have code that adds a handler to Promise UnhandledException event.  With that in place, given this simple promise chain:
```
Promise.Resolved()
    .Then(() => {
        throw new Exception();
    })
    .Catch(ex => { })
    .Done();
```
I was expecting that the UnhandledException handler would fire because I call Catch, but that appears to not be the case; my UnhandledException handler is fired whether or not I include Catch in the chain.  Should the exception stop at Catch?  Or will the UnhandledException event always get fired?

I am using version 1.3.0, compiled from GitHub source.

**Comment by RoryDungan** [comment]
Thanks, yeah that is a bug. The UnhandledException handler should only fire if you have not already handled the exception via a .Catch.

I noticed fairly recently that our test for `handled_exception_is_not_propagated_via_event()` is actually comparing the number of events raised against 1 instead of 0. The fix has been implemented and is part of pull request #52, which should be being merged soon. I'll close this issue once it is merged.

Thanks for reporting it to us and good find.
